bin/mail/2017-01-12-csp-csd-interested-teachers/generate-list

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+require_relative '../../../dashboard/config/environment'
+
+# Contacts are in a CSV exported from google-form responses.
+# Filter to send only to CSP and CSD interested teachers (not CSF),
+# and exclude teachers who have already submitted teacher applications.
+# This will be run again after new responses come in, so exclude duplicate emails
+
+IMPORT_CSV_FILENAME = 'form_responses.csv'
+TEMPLATE_NAME = '2017-01-05-go-beyond-an-hour'
+
+COL_FIRST_NAME = 'First Name'
+COL_LAST_NAME = 'Last Name'
+COL_EMAIL = 'Email'
+COL_COURSES = 'Courses'
+
+previous_contacts = []
+message_id = PEGASUS_DB[:poste_messages].where(name: TEMPLATE_NAME).map(:id).first
+previous_contacts << PEGASUS_DB[:poste_deliveries].where(message_id: message_id).map(:contact_email) if message_id
+
+count = 0
+CSV.open('contacts.csv', 'wb') do |results|
+  # headers
+  results << [:email, :name]
+
+  CSV.foreach(IMPORT_CSV_FILENAME, headers: true) do |row|
+    first_name = row[COL_FIRST_NAME]
+    last_name = row[COL_LAST_NAME]
+    email = row[COL_EMAIL].downcase
+    courses = row[COL_COURSES]
+
+    next unless email
+    next if UNSUBSCRIBERS.include? email
+    next if previous_contacts.include? email
+    next unless courses.include?('CSP') || courses.include?('CSD')
+    next if Pd::TeacherApplication.where(primary_email: email).or(Pd::TeacherApplication.where(secondary_email: email)).exists?
+
+    full_name = "#{first_name} #{last_name}"
+    results << [email, full_name]
+    count += 1
+  end
+end
+
+puts "#{count} contacts exported."
```

Comment:
Hmm, actually this doesn't work. There is no `length` property on CSV, and I get an error trying to access `results.count`:
```
/Users/andrew/.rbenv/versions/2.2.3/lib/ruby/2.2.0/csv.rb:1798:in `gets': not opened for reading (IOError)
	from /Users/andrew/.rbenv/versions/2.2.3/lib/ruby/2.2.0/csv.rb:1798:in `block in shift'
	from /Users/andrew/.rbenv/versions/2.2.3/lib/ruby/2.2.0/csv.rb:1796:in `loop'
	from /Users/andrew/.rbenv/versions/2.2.3/lib/ruby/2.2.0/csv.rb:1796:in `shift'
	from /Users/andrew/.rbenv/versions/2.2.3/lib/ruby/2.2.0/csv.rb:1738:in `each'
	from generate-list:45:in `count'
```

So I could either:
 1. count separately as before
 2. dump the output of `wc -l contacts.csv`
 3. initially collect all the results in an array, print the length, then dump to csv
 4. don't print the count at all.

Preference?